### PR TITLE
make java-resource-paths file find strategy ignore folders

### PR DESCRIPTION
**Pull request by noncom**
Fix for https://github.com/laurentpetit/ccw/issues/847

### CODE REVIEW

**Comment by laurentpetit** [comment]
Thanks for the PR!

Can you please add a line to the ChangeLog file. You'll see lots of examples in there for referencing the github issue, etc.

**Comment by laurentpetit** [comment]
Also, for commit message, we stick to the anguar project convention. I've added a line on this in the CONTRIBUTING guide: https://github.com/laurentpetit/ccw/blob/master/CONTRIBUTING.adoc#issue-reporting

File: ccw.core/src/clj/ccw/eclipse.clj [hunk 1]
```diff
@@ -489,7 +489,7 @@
                        (-> root-path (.append path))
                        (-> root-path (.append package-path) (.append path)))
                 search (workspace-resource path)]
-          :when search]
+          :when (and search (->> search (instance? org.eclipse.core.internal.resources.Folder) not))]
```
Comment (laurentpetit) [inline ccw.core/src/clj/ccw/eclipse.clj]: When possible(*), we do not depend on internal Eclipse packages. The fact that we are in presence of an Eclipse IFile is idiomatically checked through the `IResource` interface like this: `(= org.eclipse.core.resources.IResource/FILE (.getType search))`. Can you change it please?


(*): it's not always possible

**Comment by noncom** [comment]
Ooops :) sorry! I did not think of all this :) Now I followed what you're saying and now it looks more suitable. Though I am not sure if the commit message and the ChangeLog entry are correct.. doing this for the first time and also never heard of the Angular JS commit formulation convention before, but I like it :) Could you please take a look?

A technical question: why do you tend to avoid the Eclipse internal package?

**Comment by arichiardi** [comment]
Usually the internal packages are implementazione details that one should avoid relying upon...they might change very frequenty..or at least this is what experience has thought me so far :smile:

**Comment by laurentpetit** [comment]
Can you please squash these 2 commits into a single one?
Also, can you add (if missing) an import for the `IResource` class, and just use `IResource` in the code?

**Comment by noncom** [comment]
Seems done! Now I know a little more of Git :)

**Comment by laurentpetit** [comment]
Cool thanks !

**Comment by laurentpetit** [comment]
Done, your first PR is integrated ! Congrats and welcome on board !

**Comment by noncom** [comment]
Thank you! Glad we made it working! It is really cool to take part.

**Comment by arichiardi** [comment]
:balloon:

**Pull request merged**